xbmc/interfaces/AnnouncementManager.cpp

```diff
@@ -248,5 +273,24 @@ void CAnnouncementManager::Announce(AnnouncementFlag flag, const char *sender, c
   if (id > 0)
     object["item"]["id"] = id;
 
-  Announce(flag, sender, message, object);
+  DoAnnounce(flag, sender, message, object);
+}
+
+void CAnnouncementManager::Process()
+{
+  SetPriority(GetMinPriority());
+
+  while (!m_bStop)
+  {
+    if (!m_announcementQueue.empty())
+    {
+      CSingleLock lock (m_critSection);
+      auto announcement = m_announcementQueue.front();
+      m_announcementQueue.pop_front();
+      lock.Leave();
+
+      DoAnnounce(announcement.flag, announcement.sender.c_str(), announcement.message.c_str(), announcement.item, announcement.data);
+    }
+    Sleep(50);
```

Comment:
That's not a very good idea/lazy solution. It means context switching a thread that rarely do anything every 50ms. It should really be done the hard way with conditions.